### ISSUE TITLE
Ink can't compile file Unity 2018.2.0f2

### ISSUE THREAD

**Issue by afk-mcz** [inkle/ink-unity-integration#71]
Unity 2018.2.0f2
ink version: 0.8.1

Here is the error log, is the same for my Story, but it fails even with the Default example Story.

```
Unhandled error(s) occurred compiling Ink file 'Assets/Plugins/Ink/Example/Ink/story.ink'! Please report following error(s) as a bug:
﻿

UnityEngine.Debug:LogError(Object)
Ink.UnityIntegration.InkCompiler:DelayedComplete() (at Assets/Plugins/Ink/Editor/Compiler/InkCompiler.cs:320)
Ink.UnityIntegration.InkCompiler:Update() (at Assets/Plugins/Ink/Editor/Compiler/InkCompiler.cs:64)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()

```

**Comment by tomkail** [comment]
Ooh not good! Can you tell me what os, unity version and ink version you’re
using?

**Comment by afk-mcz** [comment]
Sure! 

- macOS High Sierra:  10.13.4
- Unity: 2018.2.0f2
- Ink:  0.8.1

**Comment by KumoKairo** [comment]
@afk-mcz have you managed to find the problem or you still need help?

**Comment by joshrickert** [comment]
I'm also unable to compile on Unity 2018.2.0f2, might be related.

macOS High Sierra 10.13.6
Unity 2018.2.0f2
Ink 0.8.1

Here is the full dump from my `Editor.log` when I select "Recompile All"

```

Refresh: detecting if any assets need to be imported or removed ...
Refresh: elapses 0.060194 seconds (Nothing changed)
Refresh completed in 0.060314 seconds.
Ink compilation started at 12:42:42 PM
Files compiled:
Assets/Ink/Initialize.ink
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:CompileInk(InkFile[]) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:209)
Ink.UnityIntegration.InkEditorUtils:RecompileAll() (at Assets/3rd-Party/Ink/Editor/Tools/InkEditorUtils.cs:71)
Ink.UnityIntegration.InkLibraryEditor:OnInspectorGUI() (at Assets/3rd-Party/Ink/Editor/Ink Library/InkLibraryEditor.cs:24)
UnityEditor.InspectorWindow:DrawEditor(Editor[], Int32, Boolean, Boolean&, Rect&) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1374)
UnityEditor.InspectorWindow:DrawEditors(Editor[]) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1141)
UnityEditor.InspectorWindow:OnGUI() (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:481)
System.Reflection.MonoMethod:InternalInvoke(MonoMethod, Object, Object[], Exception&)
System.Reflection.MonoMethod:Invoke(Object, BindingFlags, Binder, Object[], CultureInfo)
System.Reflection.MethodBase:Invoke(Object, Object[])
UnityEditor.HostView:Invoke(String, Object) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:295)
UnityEditor.HostView:Invoke(String) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:288)
UnityEditor.HostView:InvokeOnGUI(Rect) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:255)
UnityEditor.DockArea:OldOnGUI() (at /Users/builduser/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:390)
UnityEngine.Experimental.UIElements.IMGUIContainer:DoOnGUI(Event, Boolean) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:215)
UnityEngine.Experimental.UIElements.IMGUIContainer:HandleIMGUIEvent(Event) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:380)
UnityEngine.Experimental.UIElements.IMGUIContainer:HandleEvent(EventBase) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:359)
UnityEngine.Experimental.UIElements.EventDispatcher:DispatchEvent(EventBase, IPanel) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/EventDispatcher.cs:260)
UnityEngine.Experimental.UIElements.UIElementsUtility:DoDispatch(BaseVisualElementPanel) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:243)
UnityEngine.Experimental.UIElements.UIElementsUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:74)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:171)
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 209)

gstr.c:876: assertion 's2 != NULL' failed

Hashing assets (1 files)... 0.001 seconds
  file read: 0.000 seconds (0.002 MB)
  wait for write: 0.000 seconds (I/O thread blocked by consumer, aka CPU bound)
  wait for read: 0.000 seconds (CPUT thread waiting for I/O thread, aka disk bound)
  hash: 0.000 seconds
Updating Assets/InkLibrary.asset - GUID: 7a887ebf334124c9b84086107c350bfb...
 done. [Time: 3.298125 ms] 
gstr.c:876: assertion 's2 != NULL' failed
Refreshing native plugins compatible for Editor in 0.67 ms, found 5 plugins.
Preloading 2 native plugins for Editor in 1.52 ms.

----- Total AssetImport time: 0.439518s, AssetImport time: 0.124685s, Asset hashing: 0.000903s [1.7 KB, 1.861913 mb/s]

Ink compilation started at 12:42:43 PM
Files compiled:
Assets/Ink/Intro.ink
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:CompileInk(InkFile[]) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:209)
Ink.UnityIntegration.InkEditorUtils:RecompileAll() (at Assets/3rd-Party/Ink/Editor/Tools/InkEditorUtils.cs:71)
Ink.UnityIntegration.InkLibraryEditor:OnInspectorGUI() (at Assets/3rd-Party/Ink/Editor/Ink Library/InkLibraryEditor.cs:24)
UnityEditor.InspectorWindow:DrawEditor(Editor[], Int32, Boolean, Boolean&, Rect&) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1374)
UnityEditor.InspectorWindow:DrawEditors(Editor[]) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1141)
UnityEditor.InspectorWindow:OnGUI() (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:481)
System.Reflection.MonoMethod:InternalInvoke(MonoMethod, Object, Object[], Exception&)
System.Reflection.MonoMethod:Invoke(Object, BindingFlags, Binder, Object[], CultureInfo)
System.Reflection.MethodBase:Invoke(Object, Object[])
UnityEditor.HostView:Invoke(String, Object) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:295)
UnityEditor.HostView:Invoke(String) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:288)
UnityEditor.HostView:InvokeOnGUI(Rect) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:255)
UnityEditor.DockArea:OldOnGUI() (at /Users/builduser/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:390)
UnityEngine.Experimental.UIElements.IMGUIContainer:DoOnGUI(Event, Boolean) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:215)
UnityEngine.Experimental.UIElements.IMGUIContainer:HandleIMGUIEvent(Event) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:380)
UnityEngine.Experimental.UIElements.IMGUIContainer:HandleEvent(EventBase) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:359)
UnityEngine.Experimental.UIElements.EventDispatcher:DispatchEvent(EventBase, IPanel) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/EventDispatcher.cs:260)
UnityEngine.Experimental.UIElements.UIElementsUtility:DoDispatch(BaseVisualElementPanel) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:243)
UnityEngine.Experimental.UIElements.UIElementsUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:74)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:171)
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 209)

gstr.c:876: assertion 's2 != NULL' failed
﻿
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:ProcessError(Process, String) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:336)
Ink.UnityIntegration.InkCompiler:OnProcessError(Object, DataReceivedEventArgs) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:313)
System.Diagnostics.Process:ErrorReadNotifyUser(String)
System.Diagnostics.AsyncStreamReader:FlushMessageQueue()
System.Diagnostics.AsyncStreamReader:ReadBuffer(IAsyncResult)
System.IO.ReadWriteTask:InvokeAsyncCallback(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.IO.ReadWriteTask:System.Threading.Tasks.ITaskCompletionAction.Invoke(Task)
System.Threading.Tasks.Task:FinishContinuations()
System.Threading.Tasks.Task:FinishStageThree()
System.Threading.Tasks.Task:FinishStageTwo()
System.Threading.Tasks.Task:Finish(Boolean)
System.Threading.Tasks.Task:ExecuteWithThreadLocal(Task&)
System.Threading.Tasks.Task:ExecuteEntry(Boolean)
System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
System.Threading.ThreadPoolWorkQueue:Dispatch()
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 336)

False
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:ProcessError(Process, String) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:337)
Ink.UnityIntegration.InkCompiler:OnProcessError(Object, DataReceivedEventArgs) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:313)
System.Diagnostics.Process:ErrorReadNotifyUser(String)
System.Diagnostics.AsyncStreamReader:FlushMessageQueue()
System.Diagnostics.AsyncStreamReader:ReadBuffer(IAsyncResult)
System.IO.ReadWriteTask:InvokeAsyncCallback(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.IO.ReadWriteTask:System.Threading.Tasks.ITaskCompletionAction.Invoke(Task)
System.Threading.Tasks.Task:FinishContinuations()
System.Threading.Tasks.Task:FinishStageThree()
System.Threading.Tasks.Task:FinishStageTwo()
System.Threading.Tasks.Task:Finish(Boolean)
System.Threading.Tasks.Task:ExecuteWithThreadLocal(Task&)
System.Threading.Tasks.Task:ExecuteEntry(Boolean)
System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
System.Threading.ThreadPoolWorkQueue:Dispatch()
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 337)

65279
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:ProcessError(Process, String) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:338)
Ink.UnityIntegration.InkCompiler:OnProcessError(Object, DataReceivedEventArgs) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:313)
System.Diagnostics.Process:ErrorReadNotifyUser(String)
System.Diagnostics.AsyncStreamReader:FlushMessageQueue()
System.Diagnostics.AsyncStreamReader:ReadBuffer(IAsyncResult)
System.IO.ReadWriteTask:InvokeAsyncCallback(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.IO.ReadWriteTask:System.Threading.Tasks.ITaskCompletionAction.Invoke(Task)
System.Threading.Tasks.Task:FinishContinuations()
System.Threading.Tasks.Task:FinishStageThree()
System.Threading.Tasks.Task:FinishStageTwo()
System.Threading.Tasks.Task:Finish(Boolean)
System.Threading.Tasks.Task:ExecuteWithThreadLocal(Task&)
System.Threading.Tasks.Task:ExecuteEntry(Boolean)
System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
System.Threading.ThreadPoolWorkQueue:Dispatch()
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 338)


Hashing assets (1 files)... 0.000 seconds
  file read: 0.000 seconds (0.002 MB)
  wait for write: 0.000 seconds (I/O thread blocked by consumer, aka CPU bound)
  wait for read: 0.000 seconds (CPUT thread waiting for I/O thread, aka disk bound)
  hash: 0.000 seconds
Updating Assets/InkLibrary.asset - GUID: 7a887ebf334124c9b84086107c350bfb...
 done. [Time: 5.695718 ms] 
gstr.c:876: assertion 's2 != NULL' failed
Refreshing native plugins compatible for Editor in 0.83 ms, found 5 plugins.
Preloading 2 native plugins for Editor in 1.73 ms.

----- Total AssetImport time: 0.446714s, AssetImport time: 0.140961s, Asset hashing: 0.000297s [2.2 KB, 7.257668 mb/s]

Recompile All will compile Initialize.ink, Intro.ink
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkEditorUtils:RecompileAll() (at Assets/3rd-Party/Ink/Editor/Tools/InkEditorUtils.cs:78)
Ink.UnityIntegration.InkLibraryEditor:OnInspectorGUI() (at Assets/3rd-Party/Ink/Editor/Ink Library/InkLibraryEditor.cs:24)
UnityEditor.InspectorWindow:DrawEditor(Editor[], Int32, Boolean, Boolean&, Rect&) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1374)
UnityEditor.InspectorWindow:DrawEditors(Editor[]) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1141)
UnityEditor.InspectorWindow:OnGUI() (at /Users/builduser/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:481)
System.Reflection.MonoMethod:InternalInvoke(MonoMethod, Object, Object[], Exception&)
System.Reflection.MonoMethod:Invoke(Object, BindingFlags, Binder, Object[], CultureInfo)
System.Reflection.MethodBase:Invoke(Object, Object[])
UnityEditor.HostView:Invoke(String, Object) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:295)
UnityEditor.HostView:Invoke(String) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:288)
UnityEditor.HostView:InvokeOnGUI(Rect) (at /Users/builduser/buildslave/unity/build/Editor/Mono/HostView.cs:255)
UnityEditor.DockArea:OldOnGUI() (at /Users/builduser/buildslave/unity/build/Editor/Mono/GUI/DockArea.cs:390)
UnityEngine.Experimental.UIElements.IMGUIContainer:DoOnGUI(Event, Boolean) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:215)
UnityEngine.Experimental.UIElements.IMGUIContainer:HandleIMGUIEvent(Event) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:380)
UnityEngine.Experimental.UIElements.IMGUIContainer:HandleEvent(EventBase) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:359)
UnityEngine.Experimental.UIElements.EventDispatcher:DispatchEvent(EventBase, IPanel) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/EventDispatcher.cs:260)
UnityEngine.Experimental.UIElements.UIElementsUtility:DoDispatch(BaseVisualElementPanel) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:243)
UnityEngine.Experimental.UIElements.UIElementsUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:74)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:171)
 
(Filename: Assets/3rd-Party/Ink/Editor/Tools/InkEditorUtils.cs Line: 78)

﻿
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:ProcessError(Process, String) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:336)
Ink.UnityIntegration.InkCompiler:OnProcessError(Object, DataReceivedEventArgs) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:313)
System.Diagnostics.Process:ErrorReadNotifyUser(String)
System.Diagnostics.AsyncStreamReader:FlushMessageQueue()
System.Diagnostics.AsyncStreamReader:ReadBuffer(IAsyncResult)
System.IO.ReadWriteTask:InvokeAsyncCallback(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.IO.ReadWriteTask:System.Threading.Tasks.ITaskCompletionAction.Invoke(Task)
System.Threading.Tasks.Task:FinishContinuations()
System.Threading.Tasks.Task:FinishStageThree()
System.Threading.Tasks.Task:FinishStageTwo()
System.Threading.Tasks.Task:Finish(Boolean)
System.Threading.Tasks.Task:ExecuteWithThreadLocal(Task&)
System.Threading.Tasks.Task:ExecuteEntry(Boolean)
System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
System.Threading.ThreadPoolWorkQueue:Dispatch()
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 336)

False
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:ProcessError(Process, String) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:337)
Ink.UnityIntegration.InkCompiler:OnProcessError(Object, DataReceivedEventArgs) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:313)
System.Diagnostics.Process:ErrorReadNotifyUser(String)
System.Diagnostics.AsyncStreamReader:FlushMessageQueue()
System.Diagnostics.AsyncStreamReader:ReadBuffer(IAsyncResult)
System.IO.ReadWriteTask:InvokeAsyncCallback(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.IO.ReadWriteTask:System.Threading.Tasks.ITaskCompletionAction.Invoke(Task)
System.Threading.Tasks.Task:FinishContinuations()
System.Threading.Tasks.Task:FinishStageThree()
System.Threading.Tasks.Task:FinishStageTwo()
System.Threading.Tasks.Task:Finish(Boolean)
System.Threading.Tasks.Task:ExecuteWithThreadLocal(Task&)
System.Threading.Tasks.Task:ExecuteEntry(Boolean)
System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
System.Threading.ThreadPoolWorkQueue:Dispatch()
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 337)

65279
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Ink.UnityIntegration.InkCompiler:ProcessError(Process, String) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:338)
Ink.UnityIntegration.InkCompiler:OnProcessError(Object, DataReceivedEventArgs) (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:313)
System.Diagnostics.Process:ErrorReadNotifyUser(String)
System.Diagnostics.AsyncStreamReader:FlushMessageQueue()
System.Diagnostics.AsyncStreamReader:ReadBuffer(IAsyncResult)
System.IO.ReadWriteTask:InvokeAsyncCallback(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.IO.ReadWriteTask:System.Threading.Tasks.ITaskCompletionAction.Invoke(Task)
System.Threading.Tasks.Task:FinishContinuations()
System.Threading.Tasks.Task:FinishStageThree()
System.Threading.Tasks.Task:FinishStageTwo()
System.Threading.Tasks.Task:Finish(Boolean)
System.Threading.Tasks.Task:ExecuteWithThreadLocal(Task&)
System.Threading.Tasks.Task:ExecuteEntry(Boolean)
System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
System.Threading.ThreadPoolWorkQueue:Dispatch()
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 338)

Unhandled error(s) occurred compiling Ink file 'Assets/Ink/Initialize.ink'! Please report following error(s) as a bug:
﻿

UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
Ink.UnityIntegration.InkCompiler:DelayedComplete() (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:369)
Ink.UnityIntegration.InkCompiler:Update() (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:74)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at /Users/builduser/buildslave/unity/build/Editor/Mono/EditorApplication.cs:191)
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 369)

Unhandled error(s) occurred compiling Ink file 'Assets/Ink/Intro.ink'! Please report following error(s) as a bug:
﻿

UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
Ink.UnityIntegration.InkCompiler:DelayedComplete() (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:369)
Ink.UnityIntegration.InkCompiler:Update() (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:74)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at /Users/builduser/buildslave/unity/build/Editor/Mono/EditorApplication.cs:191)
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 369)

Ink compilation completed with errors at 12:42:44 PM
Files compiled:
Assets/Ink/Initialize.ink (With unhandled error)
Assets/Ink/Intro.ink (With unhandled error)
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogWarning(Object)
Ink.UnityIntegration.InkCompiler:DelayedComplete() (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:424)
Ink.UnityIntegration.InkCompiler:Update() (at Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs:74)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at /Users/builduser/buildslave/unity/build/Editor/Mono/EditorApplication.cs:191)
 
(Filename: Assets/3rd-Party/Ink/Editor/Compiler/InkCompiler.cs Line: 424)

gstr.c:876: assertion 's2 != NULL' failed

Hashing assets (1 files)... 0.000 seconds
  file read: 0.000 seconds (0.001 MB)
  wait for write: 0.000 seconds (I/O thread blocked by consumer, aka CPU bound)
  wait for read: 0.000 seconds (CPUT thread waiting for I/O thread, aka disk bound)
  hash: 0.000 seconds
Updating Assets/InkLibrary.asset - GUID: 7a887ebf334124c9b84086107c350bfb...
 done. [Time: 2.920477 ms] 
gstr.c:876: assertion 's2 != NULL' failed
Refreshing native plugins compatible for Editor in 0.77 ms, found 5 plugins.
Preloading 2 native plugins for Editor in 1.61 ms.

----- Total AssetImport time: 0.431444s, AssetImport time: 0.123538s, Asset hashing: 0.000383s [1.2 KB, 3.142435 mb/s]

Refresh: detecting if any assets need to be imported or removed ...

Hashing assets (1 files)... 0.000 seconds
  file read: 0.000 seconds (0.025 MB)
  wait for write: 0.000 seconds (I/O thread blocked by consumer, aka CPU bound)
  wait for read: 0.000 seconds (CPUT thread waiting for I/O thread, aka disk bound)
  hash: 0.000 seconds
----- Compute hash(es) for 1 asset(s).
----- Asset named Assets/Ink/Intro.json is skipped as no actual change.

Refresh: elapses 0.264435 seconds (Nothing changed)
Refreshing native plugins compatible for Editor in 0.64 ms, found 5 plugins.
Preloading 2 native plugins for Editor in 1.48 ms.

----- Total AssetImport time: 0.202331s, AssetImport time: 0.000000s, Asset hashing: 0.000451s [26.0 KB, 56.171059 mb/s]

Refresh completed in 0.268226 seconds.
```

**Comment by afk-mcz** [comment]
Same, I think Unity 2018 is the problem

**Comment by tomkail** [comment]
I can't reproduce this on any version of 2018. Did you download via the
asset store or github?

**Comment by tomkail** [comment]
Hmm, your line numbers don't line up with mine at all, so I think you're on
an older version. I'm going to push up a new unity package today, should be
approved by mid next week.
The other thing I've noticed is that your error message is actually showing
a byte-order-mark character which should have been stripped out, so I think
the error isn't real. Can you tell me if your files have actually compiled
(look for a JSON file next to your ink file)?

**Comment by joshrickert** [comment]
The files were compiling inconsistently, sometimes I'd get one json file, sometimes I'd get both (two ink files to compile in my project). Often none. Always the same error.

Good news though! I integrated your change in 0891a696ee775809430a98f719aed1420f080c17 and everything is working again for me. Thank you! :)

I also realized my line numbers weren't matching up because my editor had auto-formatted the file after a hotfix I applied a while back.
